kernel/mmanutils.c

```diff
@@ -536,6 +579,52 @@ int myst_release_process_mappings(pid_t pid)
     return ret;
 }
 
+int proc_pid_maps_vcallback(myst_buf_t* vbuf)
+{
+    int ret = 0;
+    pid_t pid = myst_getpid();
+    struct locals
+    {
+        char maps_entry[PATH_MAX];
+    }* locals = NULL;
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    myst_buf_clear(vbuf);
+
+    myst_spin_lock(&_mappings_lock);
+    {
+        for (myst_process_mapping_t* p = _mappings; p; p = p->next)
```

Comment:
Does the implementation support reading  portion of the /proc/pid/maps file or reading form an file offset?

Reply:
Yes `lseek` or `pread` both should work. The `vbuf` is populated at `open` and released at `close`.